apps/server/src/modules/tldraw/domain/ws-shared-doc.do.ts

```diff
@@ -0,0 +1,65 @@
+import { Doc } from 'yjs';
+import WebSocket from 'ws';
+import { Awareness, encodeAwarenessUpdate } from 'y-protocols/awareness';
+import { encoding } from 'lib0';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { WSMessageType } from '@src/modules/tldraw/types/connection-enum';
+import { TldrawWsService } from '@src/modules/tldraw/service';
+
+// disable gc when using snapshots!
+const gcEnabled: boolean = Configuration.get('TLDRAW__GC_ENABLED') as boolean;
+
+export class WsSharedDocDo extends Doc {
+	name: string;
+
+	public conns: Map<WebSocket, Set<number>>;
+
+	public awareness: Awareness;
+
+	/**
+	 * @param {string} name
+	 * @param {TldrawWsService} tldrawService
+	 */
+	constructor(name: string, private tldrawService: TldrawWsService) {
+		super({ gc: gcEnabled });
+		this.name = name;
+		this.conns = new Map();
+		this.awareness = new Awareness(this);
+		this.awareness.setLocalState(null);
+
+		this.awareness.on('update', this.awarenessChangeHandler);
+		this.on('update', (update: Uint8Array, origin, doc: WsSharedDocDo) => {
+			this.tldrawService.updateHandler(update, origin, doc);
+		});
+	}
+
+	/**
+	 * @param {{ added: Array<number>, updated: Array<number>, removed: Array<number> }} changes
+	 * @param {WebSocket | null} conn Origin is the connection that made the change
+	 */
+	public awarenessChangeHandler = (
+		{ added, updated, removed }: { added: Array<number>; updated: Array<number>; removed: Array<number> },
+		conn: WebSocket | null
+	) => {
+		const changedClients = added.concat(updated, removed);
+		if (conn !== null) {
+			const connControlledIDs = this.conns.get(conn) as Set<number>;
```

Comment:
We should avoid any as cast if possible. If not you must add a Helper, or private methods that should be include following:
1. Add a ts type guard. (input: unkown) input is Set<number> {} this method must return a boolean with true if the input format match. You must implement marker to validate this.
2. Add a XXXcastToSetNumber method that use the ts guard method with if (yes) than input as Set<number> else throw